src/prototree/optim.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from dataclasses import dataclass
+from typing import Literal
+
+import torch
+import torch.optim
+import torch.utils.data
+from torch.optim import Optimizer
+from torch.optim.lr_scheduler import MultiStepLR
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class NonlinearOptimParams:
```

Comment:
I now have an issue for this https://github.com/appliedAI-Initiative/ProtoTree/issues/34. I hadn't done it in this PR since so far I've mostly only moved the core parts of the code to Lightning.

Reply:
Sure, let's address this in follow-up work. It's not urgent